sstables/generation_type.hh

```diff
@@ -41,6 +41,14 @@ constexpr generation_type::int_t generation_value(generation_type generation) {
     return generation.value();
 }
 
+inline generation_type new_generation(std::optional<generation_type> prev = std::nullopt) {
+    if (!prev) {
+        return generation_type(1);
+    } else {
+        return generation_type(prev->value() + 1);
```

Comment:
Don't we usually increment by smp::count?

Reply:
We can but today that's done in replica::table.
The tests increment by one (generally speaking).

Reply:
ok. But what happens to the code in replica::table? Is it still using ints?

Perhaps we should increment by smp::count here too, if the code is intended to switch over.